### DISCUSSION TITLE
[V6] How to change url outside react tree?

### DISCUSSION THREAD

**Discussion by auwtch** [remix-run/react-router#8234] [Q&A]
In V5, I just used history.push, but how to do it with v6?

**Comment by wuifdesign** [comment]
having the same issue. i used it `createBrowserHistory` and passed it to `Router` the  but with v6 it seams like the history is created inside the component and not exposed.

**Reply by auwtch** [reply]
a current hotfix
```ts
export function App() {
  const navigate = useNavigate()

  // FIXME: figure out a proper way to do it
  window.__fixme__navigate = navigate
}
```

**Reply by wuifdesign** [reply]
thx, but this would be a really ugly fix. 

Is there something planed to let us navigate from outside of a react component in a "proper" way?
For example something like the `createBrowserHistory` in v5?
